testsrc/main/mondrian/rolap/NonEmptyTest.java

```diff
@@ -6133,8 +6133,6 @@ public void testMon2202SeveralFilteredHierarchiesPlusMeasureFilter() {
     }
 
     public void testMon2202AnalyzerCompoundMeasureFilterPlusTopCount() {
-        propSaver.set(
-            propSaver.properties.AlertNativeEvaluationUnsupported, "ERROR");
         assertQueryReturns(
```

Comment:
@mkambol, the reason of the test failing was not in maxConstraints. It happened because of the https://github.com/YuryBY/mondrian/blob/MONDRIAN-1803-1/src/main/mondrian/rolap/SqlContextConstraint.java#L112. Earlier it returned false without exception. I reworked the fix and now logging or an exception is provided in .isValidContext.
Additionally was added one more test scenario in the test method.